front-end/src/services/productServices.js

```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+import { setFilters, setProduct } from '../redux/productSlice';
+
+export const productService = {
+    getProducts: (dispatch, currPage, categoryOption, sizeOption, colorOption) => {
+
+        const categoryString = new URLSearchParams(categoryOption);
+        const categoryUrl = `&${categoryString.toString()}`;
+        let sizeUrl = `&size=${sizeOption.toString()}`;
+        let colorUrl = `&color=${colorOption.toString()}`;
+
+        axios.get(`//localhost:9999/products?page=${currPage}&limit=12${categoryUrl}${sizeUrl}${colorUrl}`)
```

Comment:
You can extract //localhost:9999 into variable in .env file and reuse it here.Something like that:
```suggestion
        axios.get(`${process.env.REACT_APP_BASE_URL}products?page=${currPage}&limit=12${categoryUrl}${sizeUrl}${colorUrl}`)
```

This article might be useful: [Link](https://create-react-app.dev/docs/adding-custom-environment-variables/)